entity-framework/core/managing-schemas/migrations/applying.md

```diff
@@ -0,0 +1,193 @@
+
+***
+
+Script generation accepts the following two arguments in indicate which range of migrations should be generated:
+
+* The **from** migration should be the last migration applied to the database before running the script. If no migrations have been applied, specify `0` (this is the default).
+* The **to** migration is the last migration that will be applied to the database after running the script. This defaults to the last migration in your project.
+
+## Idempotent SQL scripts
+
+The SQL scripts generated above can only be applied to change your schema from one migration to another; it is your responsibility to apply the script appropriately, and only to database in the correct migration state. EF Core also supports generating **idempotent** scripts, which internally check which migrations have already been applied (via the migrations history table), and only apply missing ones. This is useful if you don't exactly know what the last migration applied to the database was, or if you are deploying to multiple databases that may each be at a different migration.
+
+The following generates idempotent migrations:
+
+#### [.NET Core CLI](#tab/dotnet-core-cli)
+
+```dotnetcli
+dotnet ef migrations add AddNewTables --idempotent
+```
+
+#### [Visual Studio](#tab/vs)
+
+``` powershell
+Add-Migration AddNewTables -Idempotent
+```
+
+> [!WARNING]
+> Idempotent scripts have some known limitations and will not work in all scenarios. 
+
+***
+
+## Command-line tools
+
+The EF command-line tools can be used to apply migrations to a database. While productive for local development and testing of migrations, this approach isn't ideal for managing production databases:
+
+* The SQL commands are applied directly by the tool, without giving the developer a change to inspect or modify them. This can be dangerous in a production environment.
+* The .NET SDK and the EF tool must be installed on production servers.
+
+### [.NET Core CLI](#tab/dotnet-core-cli)
+
+The following updates your database to the latest migration:
+
+```dotnetcli
+dotnet ef database update
+```
+
+The following updates your database to a given migration:
+
+```dotnetcli
+dotnet ef database update AddNewTables
+```
+
+Note that this can be used to roll back to an earlier migration as well.
+
+> [!WARNING]
+> Please take note of potential data loss scenarios.
+
+### [Visual Studio](#tab/vs)
+
+The following updates your database to the latest migration:
+
+``` powershell
+Update-Database
+```
+
+The following updates your database to a given migration:
+
+``` powershell
+Update-Database AddNewTables
+```
+
+Note that this can be used to roll back to an earlier migration as well.
+
+> [!WARNING]
+> Please take note of potential data loss scenarios.
+
+***
+
+For more information on applying migrations via the command-line tools, see the [EF Core tools reference](xref:core/miscellaneous/cli/index).
+
+## Apply migrations at runtime
```

Comment:
Is there guidance on how to create a standalone executable for this that would be an interim solution until the new bundling features are ready? It seems some companies will build a console app just to host the migrations code and deploy that to run as part of CI/CD separate from the web app consuming the  database.

Reply:
I'm not specifically aware of people going to those lengths (but maybe @bricelam is)... Typically people either deploy via SQL scripts (which is what we usually recommend), or they just stick Migrate in their program startup, which is what we try to discourage...

I guess that since @bricelam is specifically working on bundles, we may want to hold off for now and just document that once it lands...